packages/repo/src/storage/memory-blockstore.ts

```diff
@@ -48,56 +62,41 @@ export class MemoryBlockstore extends RepoStorage {
     const path: CID[] = []
     while (curr !== null) {
       path.push(curr)
-      const commit = await this.get(curr, def.commit)
-      if (earliest && curr.equals(earliest)) {
-        return path.reverse()
-      }
-      const root = await this.get(commit.root, def.repoRoot)
+      const commit = await this.readObj(curr, def.commit)
+      const root = await this.readObj(commit.root, def.repoRoot)
       if (!earliest && root.prev === null) {
         return path.reverse()
+      } else if (earliest && root.prev.equals(earliest)) {
+        return path.reverse()
```

Comment:
I'd like to get some opinions on how `getCommitPath` should be ordered.

Instinct says it should be in increasing order, ie: [commitX, commitY, commitZ] (and that's how it is right now). This makes sense because, when processing commits, you want you'd probably want to do them in order

However, I've structured it similarly to `array.slice` where the first param is _inclusive_ & the second is _exclusive_
This is because if you ask for a commit path form the current commitZ to commitX, you should already have knowledge of commitX & you don't actually need it, you only need commitZ & commitY. 
And in sense, it would suggest that the you'd receive back [commitZ, commitY]

You can feel this tension on display in `pds/sync.test.ts` where I ask for `earliest: commit[2], latest: commit[15]`, but then have to compare to `commit.slice(3, 16)`. Feels weird 🤔 

You'll also notice that in `getCommitPath` on both memory-blockstore & sql-repo-storage, we have to reverse the path when returning because the path is by nature constructed from getting a reference latest and then working backwards to the earliest.

Any thoughts?

Reply:
I think you got this right as-is, and the mismatch with `slice()` makes sense to me.  This is similar but a little different than slice since we're slicing in "the opposite direction" if you can say that.  I think the case that highlights it best is `getCommitPath(x, null)` vs. `slice(x)`.  In the former it's clear we mean to ask for all commits ending at (i.e. before) `x`; in the latter we want items starting from (i.e. after) `x`. To hammer this home, we could make `earliest` optional and default it to `null`: `getCommitPath(x)` vs. `slice(x)`.